### ISSUE TITLE
`import.meta.url` should not result in `Parsing error`

### ISSUE THREAD

**Issue by vassudanagunta** [eslint/eslint#14553]
**Tell us about your environment**

- Node version: v16.0.0
- npm version: v7.11.1
- Local ESLint version: v7.25.0 (Currently used)
- Global ESLint version: Not found
- Operating System: darwin 20.3.0

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

- @typescript-eslint/parser

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "root": true,
    "parser": "@typescript-eslint/parser",
    "plugins": [
        "@typescript-eslint",
        "import",
        "babel"
    ],
    "extends": [
        "eslint:recommended",
        "plugin:@typescript-eslint/eslint-recommended",
        "plugin:@typescript-eslint/recommended"
    ],
    "rules": {
        // adapts ESLint rules to support experimental ES features.
        // See https://github.com/babel/eslint-plugin-babel for other updated rules
        "babel/semi": [2, "never"],
        "babel/new-cap": 1,
        "babel/camelcase": 1,
        "babel/no-invalid-this": 2,
        "babel/object-curly-spacing": 1,
        "babel/no-unused-expressions": 1,
        "import/no-nodejs-modules": 2,
        "@typescript-eslint/no-explicit-any": "error"
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const __dirname = path.dirname(fileURLToPath(import.meta.url))
```

```bash
eslint
```

**What did you expect to happen?**

no error for the above Javascript line. 

The [`import.meta` metaproperty proposal](https://github.com/tc39/proposal-import-meta) is currently in stage 4 of the TC39 process.  When this problem was reported a year ago (#11544), it was closed with:

> `import.meta` is a stage 3 proposal currently. ESLint only supports stage 4 proposals.

**What actually happened? Please copy-paste the actual, raw output from ESLint.**

```
  32:46  error  Parsing error: Unexpected token import

✖ 1 problem (1 error, 0 warnings)
```

**Are you willing to submit a pull request to fix this bug?**

With some guidance, and if it is not overly complex, yes.  I am unfamiliar with the codebase. If it would take a lot of time, I'd just disable this rule. Since it is a parsing error as opposed to a rule error, I suspect it should be fixed by someone who's intimate with the parser code.

**Comment by nzakas** [comment]
This is being generated by the typescript-eslint parser and not by ESLint itself. Please contact the typescript-eslint project for help.
